kubespawner/spawner.py

```diff
@@ -403,18 +403,14 @@ def _namespace_default(self):
         minlen=0,
         config=True,
         help="""
-        The command used for starting the single-user server.
+        The command used to start the single-user server.
 
-        Provide either a string or a list containing the path to the startup script command. Extra arguments,
-        other than this path, should be provided via `args`.
+        Either
+          - a string containing a single command or path to a startup script
+          - a list of the command and arguments
+          - `None` (default) to use the Docker image's `CMD`
 
-        This is usually set if you want to start the single-user server in a different python
-        environment (with virtualenv/conda) than JupyterHub itself.
-
-        Some spawners allow shell-style expansion here, allowing you to use environment variables.
-        Most, including the default, do not. Consult the documentation for your spawner to verify!
-
-        If set to `None`, Kubernetes will start the `CMD` that is specified in the Docker image being started.
+        `cmd` will be passed to the image's `ENTRYPOINT`.
```

Comment:
I think this is a problematic statement for two reasons.

1. It only influence the k8s Pod rather than the Dockerfile or similarly.
2. It actually configures the k8s container's `args` (like Dockerfile's CMD) rather than `command` (like Dockerfile's ENTRYPOINT).

A related confusion is described in https://github.com/jupyterhub/kubespawner/issues/493.

I think we can make small progressions and I find this change PR mergable if we just update this to reference that it implies setting the k8s Pod's container's `args` where the `command` won't be set. In other words, any Dockerfile's ENTRYPOINT will always be respected. Here is [related documentation about that](https://kubernetes.io/docs/tasks/inject-data-application/define-command-argument-container/#notes).

Reply:
THANK YOU FOR YOUR WORK ON THIS @manics!! :heart:

I forgot to convey that part before!